feat: support svg fill rule subpath rendering

creativoma opened the pull request:
##### Description of change

This PR fixes two critical SVG rendering issues reported in #11371:

**Issues Fixed:**
1. **Subpath order dependency**: Inner subpaths defined before outer subpaths weren't rendering holes correctly
2. **Ignored fill-rule attribute**: `fill-rule="nonzero"` was being ignored, always showing holes regardless of winding direction

**Solution:**
- **Smart hole detection**: Only processes holes when `fill-rule="evenodd"` is explicitly set AND multiple subpaths exist
- **Advanced subpath processing**: Implements area-based sorting and intelligent hole/fill pattern detection for complex SVGs
- **Backward compatibility**: Maintains original behavior for all existing cases

**Technical Implementation:**
- Enhanced `SVGParser.ts` with proper fill-rule logic and multi-subpath handling
- Added helper functions: `extractSubpaths()`, `calculatePathArea()`, `checkForNestedPattern()`
- Two-strategy hole processing: nested holes vs multiple holes approach
- Conservative approach: only applies new logic when explicitly needed

**Test Results:**
All 7 test cases now render correctly:
- ✅ Evenodd with holes works regardless of subpath order
- ✅ Nonzero with same winding shows solid shapes (no holes)
- ✅ Complex multi-hole and nested-hole patterns work correctly
- ✅ Real-world SVGs (like PixiJS logo) render perfectly
- ✅ Single subpath stars show correct evenodd vs nonzero behavior

**Closes #11371**

##### Pre-Merge Checklist

- [x] Tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

bigtimebuddy commented on src/scene/graphics/shared/svg/SVGParser.ts:
```diff
@@ -245,3 +317,122 @@ function renderChildren(svg: SVGElement, session: Session, fillStyle: FillStyle,
+    const largestToSecondRatio = largestArea / secondArea;
+    const secondToSmallestRatio = secondArea / smallestArea;
+
+    if (largestToSecondRatio > 3 && secondToSmallestRatio < 2)
```
What are the significance of these magic numbers?

creativoma commented:
Hi @bigtimebuddy!
Added comprehensive visual tests for all SVG fill-rule scenarios and applied your code review suggestions (destructuring + documented magic numbers 😊)

bigtimebuddy reviewed (approved):
Nice!

creativoma commented:
Hey @GoodBoyDigital! Thanks for the feedback! 🙌

You're absolutely right - I've refactored those utility functions into organized files under the `utils` folder:

- **`pathOperations.ts`** → `extractSubpaths`, `calculatePathArea`, `appendSVGPath`  
- **`fillOperations.ts`** → `checkForNestedPattern`, `getFillInstructionData`

The `SVGParser.ts` is now much cleaner and focused on its core parsing logic! Also added comprehensive unit tests for all the utility functions to ensure they work correctly in isolation.

All tests pass and linting is clean.
